### DISCUSSION TITLE
Generate a non-HTML file from `DIRECT_TEMPLATES`?

### DISCUSSION THREAD

**Discussion by plgruener** [getpelican/pelican#3402] [Q&A]
Is it possible to use the direct templates to generate a non-html file? Specifically I want to generate a .csv with some metadata (title, author etc.) of all my articles.  

Writing the jinja template and appending it to DIRECT_TEMPLATES works, but the resulting file is always a `.html`. How can I get it to be a `.csv`? I tried changing the path of the output file via a `save_as=…` into EXTRA_PATH_METADATA, but that did not work.  
Looking at the code in `generators.py`, it seems like the `.html` extension is hardcoded and can't be changed.  
(If that is indeed the case, I'd offer writing a PR changing that (eg. based on the extension of the corresponding jinja template) if you'd like.)

**Comment by justinmayer** [comment]
This is what the docs say about `DIRECT_TEMPLATES`:

> Typically direct templates are used to generate index pages for collections of content (e.g., category and tag index pages).

Based on that description, I get the impression your intended use case was not in scope when this setting was developed.

The combination of [`TEMPLATE_PAGES`](https://docs.getpelican.com/en/latest/settings.html#template-pages) and `TEMPLATE_EXTENSIONS` should be able to produce files with arbitrary extensions. Perhaps those settings might provide a path toward achieving what you are trying to accomplish?
